Sources/AST/Environment.swift

```diff
@@ -98,6 +98,22 @@ public struct Environment {
     return declaredStructs.contains { $0.name == type }
   }
 
+  /// Whether a struct is self referencing.
+  public func isStructSelfReferencing(root type: RawTypeIdentifier, parent: RawTypeIdentifier) -> PropertyInformation? {
```

Comment:
I think this function should be named something else, as it doesn't return a boolean. What about `selfReferentialProperty(in type: RawTypeIdentifier, enclosingType: RawTypeIdentifier)`?